baseapp/server.go

```diff
@@ -130,6 +135,31 @@ func (s *Server) Start() error {
 	return s.server.ListenAndServe()
 }
 
+// Start starts the server and blocks.
+func (s *Server) Start() error {
+	// maintain backwards compatibility
+	if s.config.ShutdownWaitTime == nil {
+		return s.start()
+	}
+
+	go func() {
+		if err := s.start(); err != nil {
+			s.logger.Error().Err(err).Msg("Shutting down server")
```

Comment:
The docs say that `ListendAndServe` immediately returns `http.ErrServerClosed` once shutdown is called. It would be nice to ignore that error as expected and only log other unexpected errors.

We should also make sure the server process exits if it never starts (say due to port conflict or invalid certificates). Right now, I think the error would be logged and then the process would block waiting for a signal. Maybe add an error channel in addition to the interrupt one and take different action based on which one receives a message first?